cosmic-core/engine/schema/src/main/java/com/cloud/vm/dao/DomainRouterDaoImpl.java

```diff
@@ -357,7 +357,7 @@ public DomainRouterVO persist(final DomainRouterVO router, final List<Network> g
     public void addRouterToGuestNetwork(final VirtualRouter router, final Network guestNetwork) {
         if (_routerNetworkDao.findByRouterAndNetwork(router.getId(), guestNetwork.getId()) == null) {
             final NetworkOffering off = _offDao.findById(guestNetwork.getNetworkOfferingId());
-            if (!off.getName().equalsIgnoreCase(NetworkOffering.DefaultPrivateGatewayNetworkOffering)) {
+            if (!off.getGuestType().equals(Network.GuestType.Private)) {
```

Comment:
I'd replace this `if` with: `if (!Network.GuestType.Private.equals(off.getGuestType()))`